tests/integration/test_materialize_mysql_database/materialize_with_ddl.py

```diff
@@ -461,3 +462,52 @@ def select_without_columns(clickhouse_node, mysql_node, service_name):
     clickhouse_node.query("DROP VIEW v")
     clickhouse_node.query("DROP DATABASE db")
     mysql_node.query("DROP DATABASE db")
+
+
+def err_sync_user_privs_with_materialize_mysql_database(clickhouse_node, mysql_node, service_name):
+    mysql_node.query("CREATE DATABASE test_database DEFAULT CHARACTER SET 'utf8'")
+    mysql_node.query("CREATE TABLE test_database.test_table_1 (id INT NOT NULL PRIMARY KEY) ENGINE = InnoDB;")
+    mysql_node.query("INSERT INTO test_database.test_table_1 VALUES(1), (2), (3), (4), (5), (6);")
+
+    mysql_node.result("SHOW GRANTS FOR 'test'@'%';")
+
+    clickhouse_node.query(
+        "CREATE DATABASE test_database ENGINE = MaterializeMySQL('{}:3306', 'test_database', 'test', '123')".format(
+            service_name))
+
+    # wait MaterializeMySQL read binlog events
+    time.sleep(90)
+
+    assert "test_table_1" in clickhouse_node.query("SHOW TABLES FROM test_database")
```

Comment:
```suggestion
    check_query(clickhouse_node, "SHOW TABLES FROM test_database FORMAT TSV", "test_table_1\n")
```
`check_query(...)` should be enough to wait for MaterializeMySQL to read binlog events. If it's not, then we probably have some bug in replication from MySQL

Reply:
yep, I will add retry count.